domain/model/state/state.go

```diff
@@ -563,81 +520,102 @@ INSERT INTO model_secret_backend (model_uuid, secret_backend_uuid) VALUES (?, ?)
 	modelType coremodel.ModelType,
 	input model.ModelCreationArgs,
 ) error {
-	cloudStmt := `
-SELECT uuid
-FROM cloud
-WHERE name = ?
-`
-	var cloudUUID string
-	err := tx.QueryRowContext(ctx, cloudStmt, input.Cloud).Scan(&cloudUUID)
-	if errors.Is(err, sql.ErrNoRows) {
-		return fmt.Errorf("%w cloud %q", errors.NotFound, input.Cloud)
+	cloudName := dbName{Name: input.Cloud}
+
+	cloudStmt, err := preparer.Prepare(`SELECT &dbUUID.* FROM cloud WHERE name = $dbName.name`, dbUUID{}, cloudName)
+	if err != nil {
+		return errors.Annotate(err, "preparing select cloud statement")
+	}
+
+	var cloudUUID dbUUID
+	if err := tx.Query(ctx, cloudStmt, cloudName).Get(&cloudUUID); errors.Is(err, sqlair.ErrNoRows) {
+		return fmt.Errorf("%w: %q", clouderrors.NotFound, input.Cloud)
 	} else if err != nil {
 		return fmt.Errorf("getting cloud %q uuid: %w", input.Cloud, err)
 	}
 
-	userStmt := `
-SELECT uuid
-FROM user
-WHERE uuid = ?
-AND removed = false
-`
-	var userUUID string
-	err = tx.QueryRowContext(ctx, userStmt, input.Owner).Scan(&userUUID)
-	if errors.Is(err, sql.ErrNoRows) {
+	ownerUUID := dbUserUUID{UUID: input.Owner.String()}
+	userStmt, err := preparer.Prepare(`
+		SELECT &dbUserUUID.uuid
+		FROM user
+		WHERE uuid = $dbUserUUID.uuid
+		AND removed = false
+	`, ownerUUID)
+	if err != nil {
+		return errors.Annotatef(err, "preparing check user exists statement")
+	}
+	err = tx.Query(ctx, userStmt, ownerUUID).Get(&ownerUUID)
+	if errors.Is(err, sqlair.ErrNoRows) {
 		return fmt.Errorf("%w for model owner %q", accesserrors.UserNotFound, input.Owner)
 	} else if err != nil {
 		return fmt.Errorf("getting user uuid for setting model %q owner: %w", input.Name, err)
 	}
 
 	// If a model with this name/owner was previously created, clean it up
 	// before creating the new model.
-	if err := cleanupBrokenModel(ctx, tx, input.Name, input.Owner); err != nil {
+	if err := cleanupBrokenModel(ctx, preparer, tx, input.Name, input.Owner); err != nil {
 		return fmt.Errorf("deleting broken model with name %q and owner %q: %w", input.Name, input.Owner, err)
 	}
 
-	stmt := `
-INSERT INTO model (uuid,
-            cloud_uuid,
-            model_type_id,
-			life_id,
-            name,
-            owner_uuid)
-SELECT ?, ?, model_type.id, ?, ?, ?
-FROM model_type
-WHERE model_type.type = ?
-`
+	model := dbInitialModel{
+		UUID:      modelUUID.String(),
+		CloudUUID: cloudUUID.UUID,
+		ModelType: modelType.String(),
+		LifeID:    int(life.Alive),
+		Name:      input.Name,
+		OwnerUUID: input.Owner.String(),
+	}
+
+	stmt, err := preparer.Prepare(`
+		INSERT INTO model (uuid,
+		            cloud_uuid,
+		            model_type_id,
+		            life_id,
+		            name,
+		            owner_uuid)
+		SELECT  $dbInitialModel.uuid, 
+				$dbInitialModel.cloud_uuid, 
+				model_type.id, 
+				$dbInitialModel.life_id, 
+				$dbInitialModel.name,
+				$dbInitialModel.owner_uuid
+		FROM model_type
+		WHERE model_type.type = $dbInitialModel.model_type
```

Comment:
Can you add a note for when someone does the model epic, to do a select before an insert?

Reply:
What would be the reasoning for that? If its going to be a write transaction anyway might as well do it all in one hit right? The select would only fail if something was very wrong. 

It would make much cleaner code though.